src/main/native/windows/processes-jni.cc

```diff
@@ -470,18 +494,53 @@ class NativeProcess {
 
       // Any other case is an error and should be reported back to Bazel.
       default:
-        result = 2;
         error_ = bazel::windows::MakeErrorMessage(
             WSTR(__FILE__), __LINE__, L"NativeProcess:WaitFor", ToString(pid_),
             GetLastError());
-        break;
+        return 2;
     }
 
     if (stdin_ != INVALID_HANDLE_VALUE) {
       CloseHandle(stdin_);
       stdin_ = INVALID_HANDLE_VALUE;
     }
 
+    // Ensure that the process is really terminated (if WaitForSingleObject
+    // above timed out, we have to explicitly kill it) and that it doesn't
+    // leave behind any subprocesses.
+    if (!Terminate()) {
+      return 2;
```

Comment:
Could you extract these return values to named constants?

Reply:
Done.